### PR TITLE
Feature/one click copy

### PR DESCRIPTION
**Pull request by learykara**
move the component from settings.tf.com/refer-a-friend w one-click-copy to thinkful-ui

### CODE REVIEW

File: src/OneClickCopy/OneClickCopy.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            [React.PropTypes.string, React.PropTypes.object]),
+    inputText: React.PropTypes.string,
+    onCopyClick: React.PropTypes.func,
+    onCopyClickSuccess: React.PropTypes.func,
+    onCopyClickFail: React.PropTypes.func,
+  }
+
+  _handleInputClick = (event) => {
+    event.preventDefault();
+    this.refs.copyArea.select();
```
Comment (ripleyaffect) [inline src/OneClickCopy/OneClickCopy.jsx]: why this selection behavior, if it's just duplicated in the `_handleCopyToClipboard` function?

File: src/OneClickCopy/index.es6 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const {OneClickCopy} = require('./OneClickCopy');
+
+require('./one-click-copy.less');
+
+module.exports = {OneClickCopy}
```
Comment (ripleyaffect) [inline src/OneClickCopy/index.es6]: `export default require('./OneClickCopy');` does the same thing

**Comment by ripleyaffect** [comment]
:+1:

File: src/OneClickCopy/OneClickCopy.jsx (hunk 1 above)
Comment (learykara) [inline src/OneClickCopy/OneClickCopy.jsx]: if you click in the input at all, the whole thing should be selected but not editable. so this is separate from the copy function

File: src/OneClickCopy/index.es6 (hunk 2 above)
Comment (learykara) [inline src/OneClickCopy/index.es6]: will update

**Pull request merged**